dataprofiler/profilers/profile_builder.py

```diff
@@ -1531,21 +1546,24 @@ def save(self, filepath: str = None) -> None:
         :type filepath: String
         :return: None
         """
-        # Create dictionary for all metadata, options, and profile
-        data_dict = {
-            "total_samples": self.total_samples,
-            "sample": self.sample,
-            "encoding": self.encoding,
-            "file_type": self.file_type,
-            "_samples_per_update": self._samples_per_update,
-            "_min_true_samples": self._min_true_samples,
-            "_empty_line_count": self._empty_line_count,
-            "memory_size": self.memory_size,
-            "options": self.options,
-            "_profile": self.profile,
-            "times": self.times,
-        }
-        self._save_helper(filepath, data_dict)
+        if filepath is None or filepath.split(".")[-1] != "json":
```

Comment:
wonder if this should be a bit more clear... the `filepath` connotates to me just the folder path... not necessarily the name of the file ... but the filename is vital here with its extension becuause the core logic to testing wether to go with `_pkl_save_helper` or `_json_save_helper` is keying off the extension of the file

Reply:
we allow for `None`, yes; but the code in `_pkl_save_helper` auto creates a `filepath` (really it is create a filename) automatically with a timestamp for uniquness

Reply:
I think the logic we might is more like:
1. use `type` first to determine how to save
    a. basically can force save with any extension to any type 
2. if `type=None` look to extension to save
3. if extension is not clear, choose pickle
4. if filepath not specified, choose pickle

Reply:
> use type first to determine how to save

agree